internal/renderer/service_util.go

```diff
@@ -341,6 +343,51 @@ func createLbService4Gateway(c *RenderContext, gw *gwapiv1a2.Gateway) *corev1.Se
 		})
 	}
 
+	// Set health check port
+	regexProtocol := `^service\.beta\.kubernetes\.io\/.*health.*protocol$`
```

Comment:
Quick quick: the regexps are always the same, can we turn them into package globals so that we compile the regexps only once (on startup) and not on each function call. Not sure whether it's worth doing this, but the code would surely become simpler.